x/perpetual/keeper/settle_funding_fee_distribution.go

```diff
@@ -48,48 +42,53 @@ func (k Keeper) SettleFundingFeeDistribution(ctx sdk.Context, mtp *types.MTP, po
 	if mtp.Position == types.Position_LONG {
 		// Ensure liabilitiesLong is not zero to avoid division by zero
 		if liabilitiesLong.IsZero() {
-			return sdk.Coin{}, types.ErrAmountTooLow
+			return types.ErrAmountTooLow
 		}
 		fundingFeeShare = sdk.NewDecFromInt(mtp.Liabilities).Quo(sdk.NewDecFromInt(liabilitiesLong))
 		totalFund = short
 	} else {
 		// Ensure liabilitiesShort is not zero to avoid division by zero
 		if liabilitiesShort.IsZero() {
-			return sdk.Coin{}, types.ErrAmountTooLow
+			return types.ErrAmountTooLow
 		}
 		fundingFeeShare = sdk.NewDecFromInt(mtp.Liabilities).Quo(sdk.NewDecFromInt(liabilitiesShort))
 		totalFund = long
 	}
 
 	// if funding fee share is zero, skip mtp
 	if fundingFeeShare.IsZero() {
-		return sdk.Coin{}, nil
+		return nil
 	}
 
 	// calculate funding fee amount
 	fundingFeeAmount := sdk.NewCoin(baseCurrency, totalFund.Mul(fundingFeeShare).TruncateInt())
-	toPay := sdk.Coin{}
 
-	if balance.Amount.LT(fundingFeeAmount.Amount) {
-		toPay = fundingFeeAmount
-	} else {
-		// transfer funding fee amount to mtp address
-		if err := k.bankKeeper.SendCoins(ctx, fundingFeeCollectionAddress, mtpAddress, sdk.NewCoins(fundingFeeAmount)); err != nil {
-			return sdk.Coin{}, err
-		}
+	// update mtp custody
+	mtp.Custody = mtp.Custody.Add(fundingFeeAmount.Amount)
+
+	// update pool custody balance
+	err = pool.UpdateCustody(ctx, mtp.CustodyAsset, fundingFeeAmount.Amount, false, mtp.Position)
+	if err != nil {
+		return err
+	}
+
+	// update accounted balance for custody side
+	err = pool.UpdateBalance(ctx, mtp.CustodyAsset, fundingFeeAmount.Amount, false, mtp.Position)
```

Comment:
@amityadav0 the funding fee is collected here https://github.com/elys-network/elys/pull/844/files#diff-943de122eee4295bcf39ae75c877dc118541e577b8cd1ecb4b64ee2cfae9d8e6R62 through out a funding address, we need to either get rid of the funding address and keep track of an accounting at the pool level of all the funding fees that were collected that need to be distributed to the opposite trend or if we want to continue using the funding address then we should send the collected funding fees from this address to the perpetual module wallet and update the custody balance accordingly.

Reply:
got it, I will update to later approach

Reply:
after thinking more, I think we should keep it as is,
there will be a edge case when collection address won't have enough to pay to perpetual wallet, although we need to update custody and pool which will not be correct as balance won't match

As of now we handled it using toPay mechanism. I think we should close this PR

Reply:
@amityadav0 got it although I believe it would be a better UX if the funding fee is added to the custody amount rather than directly distributed to their wallet, for 2 reasons:

1) we already reduce custody for collecting the funding fee from the traders that follows market trend, so it would actually make sense to add to the custody rather than distribute to their wallet, as we are not using their wallet tokens to pay for funding fees

2) earning funding fees would allow traders to keep their position healthier and therefore afloat, that is more true for positions that go against the market trend.

Reply:
consider an example:

there are 100 positions, we need to collect from 50 and distribute to 50 as well.
and for collection let's say 10 positions have settled (As we do lazy settlement)

Now if 50 distributions happens, there won't be enough to pay from collection address but we will add it to pool right ?

This may result in error later, when closing positions or any other health related calculations.

Do you have any suggestions on this ?

Reply:
agreed the first approach allows to avoid having to move liquidity around that way we could optimize the accounting flow

Reply:
okay, I am adding that

Reply:
updated